### ISSUE TITLE
Performance gap on AIME2024

### ISSUE THREAD

**Issue by JH-ninjatech** [simplescaling/s1#41]
Hi @Muennighoff,

Thanks for sharing the repo. I have trained the Qwen2.5-32B-Instruct model following the training script using 8xH200 GPUs. My training loss curve looks similar to what reported in the paper (screenshots below). 
<img width="1208" alt="Image" src="https://github.com/user-attachments/assets/ca95be0b-8f9c-4fef-98db-80887f95de2b" />
My training loss curve:
<img width="1671" alt="Image" src="https://github.com/user-attachments/assets/70451967-7954-46cc-a6e9-3b5122ce6ff7" />

However, when I did the AIME 2024 benchmarking using my own prompt, my fine-tuned model can only achieve 9/30=30% accuracy. So my questions is: will system prompt or user prompt affect model performance on AIME by a large amount?
My system prompt and user prompt are:
```python
system_prompt = "You are a helpful assistant for solving math problem. Provide the final answer in \\boxed{} format."
user_prompt = aime_problem
```

**Comment by Muennighoff** [comment]
Yeah the prompt can affect things quite a lot - I recommend trying with the same prompt as we do and then I believe you should be able to reproduce the results.

**Comment by JH-ninjatech** [comment]
@Muennighoff thanks for the quick response. What I found was when benchmarking the Qwen2.5-32B-Instruct base model using the same system/user prompt as above, I can get 9/30=30% as well. Does it mean the finetuned model become more sensitive to the system prompt?

**Comment by Muennighoff** [comment]
Hmm maybe you are not using the same system prompt during finetuning vs after finetuning for evaluation? I could imagine it being a problem if you use a different system prompt during finetuning

**Comment by JH-ninjatech** [comment]
@Muennighoff just tried benchmarking aime2024 (30 questions) once again by fixing the system prompt following the training data. I can get 19 / 30 = 63.33% by using `simplescaling/s1.1-32B` with no Budget forcing. I am surprised that it is higher than what you reported in the paper.

**Comment by Muennighoff** [comment]
Wow did you retrain a model or you just evaluated `simplescaling/s1.1-32B` as is?

**Comment by JH-ninjatech** [comment]
as it is

**Comment by Muennighoff** [comment]
Oh interesting; Maybe you're evaling it in bf16? or maybe with transformers? I only got 56.7 with fp32 but 60.0 with bf16 I think

**Comment by JH-ninjatech** [comment]
That makes sense. I evaluated it with `bf16`.

**Comment by mucaoshen** [comment]
> Oh interesting; Maybe you're evaling it in bf16? or maybe with transformers? I only got 56.7 with fp32 but 60.0 with bf16 I think

encounter the same interesting fact when evaling it . Is it because of using bf16 type during training?

**Comment by Muennighoff** [comment]
we eval in fp32 to make results more reproducible but there are determinism issues due to vllm ([appendix a](https://arxiv.org/pdf/2501.19393#page=16.08))

**Comment by lixin2002cn** [comment]
> no re-train, just download the model weights of `simplescaling/s1.1-32B`. I am using my own evaluation script for aime2024 though. My system prompt and user prompt are:
> 
> messages=[
>     {"role": "system", "content": "You are Qwen, created by Alibaba Cloud. You are a helpful assistant."},
>     {"role": "user", "content": aime_problem},
> ]

Hi where did you set it?
